src/server/server_send.c

```
@@ -124,3 +124,63 @@ void bc_send_to_all(const u8 *payload, int payload_len, bool reliable)
+    const bc_qmsg_t *promoted[BC_DRAIN_QUEUE_CAP];
+    int promoted_count = 0;
+
+    int len = bc_drain_to_buf(&peer->drain, BC_DIR_SERVER, pkt, sizeof(pkt),
+                              promoted, BC_DRAIN_QUEUE_CAP, &promoted_count);
+    if (len <= 0) return;  /* Pass-4 gate closed / nothing to send this tick */
+
+    /* Pass 3 promoted these unreliable messages to reliable tracking on first
+     * delivery: register them so they retransmit until ACKed, matching stock
+     * burst-absorption where a drained unreliable graduates to the reliable
+     * path. */
+    for (int p = 0; p < promoted_count; p++) {
```

Clang (14.0.6)

[warning] loop variable name 'p' is too short, expected at least 2 characters

(readability-identifier-length)